app/src/main/java/com/eveningoutpost/dexdrip/Services/MissedReadingService.java

```diff
@@ -42,20 +43,26 @@ protected void onHandleIntent(Intent intent) {
         bg_missed_minutes =  Integer.parseInt(prefs.getString("bg_missed_minutes", "30"));
         otherAlertSnooze =  Integer.parseInt(prefs.getString("other_alerts_snooze", "20"));
 
-        if (bg_missed_alerts && BgReading.getTimeSinceLastReading() > (bg_missed_minutes * 1000 * 60)) {
+        if (bg_missed_alerts && BgReading.getTimeSinceLastReading() > (bg_missed_minutes * 1000 * 60)
+                &&
+                prefs.getLong("alerts_disabled_until", 0) < new Date().getTime()) {
             Notifications.bgMissedAlert(mContext);
             checkBackAfterSnoozeTime();
-        } else {
-            checkBackAfterMissedTime();
+        } else  {
+            long alarmIn = prefs.getLong("alerts_disabled_until", 0) - new Date().getTime();
+            if (alarmIn <= 0) {
+                alarmIn = Long.parseLong(prefs.getString("bg_missed_minutes", "30"))* 1000 * 60;
```

Comment:
@tzachi-dar, you might be onto something. In this case, we had the problem already before. It is the same behaviour as before.
We could fix that in a separate pr then.